Source/WTF/wtf/cocoa/LanguageCocoa.mm

```diff
@@ -73,4 +74,14 @@ bool canMinimizeLanguages()
 ALLOW_NEW_API_WITHOUT_GUARDS_END
 }
 
+void overrideUserPreferredLanguages(const Vector<String>& override)
+{
+    LOG_WITH_STREAM(Language, stream << "Languages are being overridden to: " << override);
+    NSDictionary *existingArguments = [[NSUserDefaults standardUserDefaults] volatileDomainForName:NSArgumentDomain];
+    auto newArguments = adoptNS([existingArguments mutableCopy]);
+    [newArguments setValue:createNSArray(override).get() forKey:@"AppleLanguages"];
+    [[NSUserDefaults standardUserDefaults] setVolatileDomain:newArguments.get() forName:NSArgumentDomain];
```

Comment:
This _might_ be considered to be a footgun. Because all sorts of Cocoa code consults with this NSUserDefault, it's only really meaningful to set it before NSApplicationMain is called. If you switch it out while the process is running, that's a bit scary - Cocoa code might get confused.

It might not matter at all, but I kind of feel that if we want to do this, we might want to consider transitioning existing tests which call this at runtime to use `<!-- webkit-tests-runner -->` instead.

Or maybe we could just YOLO instead

Reply:
I thought about porting tests to use `<!-- webkit-tests-runner -->` instead. However, some tests cover the language change notification. We couldn't test that when using `<!-- webkit-tests-runner -->`.

I don't think I see the issue with updating the AppleLanguages at runtime, this is exactly what happens if you change language in the OS system preferences while Safari is running. We even listen for such changes in Webkit so we can fire the languagechange event.